realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -2054,6 +2054,42 @@ private OsResults lazyFindAll() {
         return this;
     }
 
+    /**
+     * This predicate is only relevant for Query-based Realms.
+     * <p>
+     * Objects referenced through fields marked with {@link io.realm.annotations.LinkingObjects} are normally not downloaded
```

Comment:
If we have two classes `A` and `B`, and three objects `a1`, `a2`, and `b1`. If `a1` has a linking object to `b1`, and `b1` has a link to `a2`, all three objects will be included, when the query is on `A`, right?

Reply:
Good question. I would assume yes.

Reply:
I think it is important to document the behaviour. Maybe the API doc isn't the right place.

Reply:
We should probably explain it in more details here: https://docs.realm.io/sync/using-synced-realms/syncing-data#subscribing-to-data-reading-data